frontstage/views/secure_messaging.py

```diff
@@ -95,195 +48,108 @@ def message_get(session, label, message_id):
-        ru_id = associations[0].get('partyId')
-        logger.debug('Successfully received ru_id', ru_id=ru_id)
-    else:
-        logger.error('Respondent has no associations', party_id=party_id)
-        ru_id = None
-    return ru_id
-
-
-def get_collection_case(party_id):
-    url = app.config['RM_CASE_GET_BY_PARTY'].format(app.config['RM_CASE_SERVICE'], party_id, "")
-    logger.debug('Retrieving collection case id')
-    collection_response = requests.get(url, auth=app.config['BASIC_AUTH'])
-    if collection_response.status_code == 204:
-        logger.warning('No case found')
-        return None
-    elif collection_response.status_code != 200:
-        logger.error('Failed to retrieve collection case id')
-        raise ExternalServiceError(collection_response)
-    collection_response_json = collection_response.json()
-    case_id = collection_response_json[0].get('id')
-    logger.debug('Successfully received collection case', case_id=case_id)
-    return case_id
-
-
-def get_survey_id(party_id):
-    url = app.config['RAS_PARTY_GET_BY_RESPONDENT'].format(app.config['RAS_PARTY_SERVICE'], party_id)
-    logger.debug('Retrieving survey id', party_id=party_id)
-    survey_response = requests.get(url, auth=app.config['BASIC_AUTH'])
-    if survey_response == 404:
-        logger.warning('No respondent found in party service', party_id=party_id)
-        return None
-    elif survey_response.status_code != 200:
-        logger.error('Failed to retrieve survey id', party_id=party_id)
-        raise ExternalServiceError(survey_response)
-    survey_response_json = survey_response.json()
-    associations = survey_response_json.get('associations')
-    if associations:
-        survey_name = associations[0].get('enrolments')[0].get('surveyId')
-    else:
-        logger.error('Respondent has no associations', party_id=party_id)
-        survey_name = None
-    logger.debug('Successfully received survey name', survey_id=survey_name, party_id=party_id)
-    return survey_name
-
-
-def get_messages_list(label, logger):
+def get_messages_list(label):
     logger.debug('Attempting to retrieve messages')
+
+    # Form api request
     headers = {"Authorization": request.cookies['authorization']}
     endpoint = app.config['GET_MESSAGES_URL']
     parameters = {"label": label} if label else {}
     response = api_call('GET', endpoint, parameters=parameters, headers=headers)
 
+    # Check for failure calling Frontstage API
     if response.status_code != 200:
-        logger.error('Error retrieving user messages')
-        raise ExternalServiceError(response)
+        logger.error('Error connecting to frontstage api')
+        raise ApiError('FA000')
 
     messages_list = json.loads(response.text)
+
+    # Handle Api Error codes
+    error_code = messages_list.get('error', {}).get('code')
+    if error_code == 'FA001':
+        logger.error('Failed to retrieve messages list')
+        raise ApiError('FA001')
+    elif error_code == 'FA002':
```

Comment:
I think... {"unread_messages_total": { "error": { "code": "FA002",}}} is returned so error_code would never be 'FA002'

Reply:
You're right, I've made the changes you suggested in frontstage api and it fixed this mistake I'm pretty sure